monitor/internal/kubernetes/cache.go

```diff
@@ -0,0 +1,147 @@
+package kubernetesmonitor
+
+import (
+	"sync"
+
+	"github.com/aporeto-inc/trireme-lib/policy"
+)
+
+type puidCacheEntry struct {
+	// podID is the reference to the Kubernetes pod that this container refers to
+	kubeIdentifier string
+
+	// The latest reference to the runtime as received from DockerMonitor
+	dockerRuntime policy.RuntimeReader
+
+	// The latest reference to the runtime as received from DockerMonitor
+	kubernetesRuntime policy.RuntimeReader
+}
+
+type podCacheEntry struct {
+	// puIDs us a map containing a link to all the containers currently known to be part of that pod.
+	puIDs map[string]bool
+}
+
+// Cache keeps all the state needed for the integration.
+type cache struct {
+	// popuidCache keeps a mapping between a PUID and the corresponding puidCacheEntry.
+	puidCache map[string]*puidCacheEntry
+
+	// podCache keeps a mapping between a POD/Namespace name and the corresponding podCacheEntry.
+	podCache map[string]*podCacheEntry
+
+	// Lock for the whole cache
+	sync.RWMutex
+}
+
```

Comment:
It is declared as a RW mutex, but below it is only used a plain mutex. No RLock() RUnlock(). I also think that if we used the sync Map package of go it would be better. Trying to migrate all caches to this. In any case, let's use RLock() in the read operations below.

Reply:
Done, Thanks, I didn't know about `Rlock` and `RUnlock`